### DISCUSSION TITLE
Whether Strimzi needs a LoadBalancer when I configure external listener with ingress type?

### DISCUSSION THREAD

**Discussion by 1123183721** [strimzi/strimzi-kafka-operator#7666] [Q&A] [answered]
T have configured external listener with ingress type. but operator reported the following log：
`2022-11-21 02:29:26 WARN  AbstractOperator:516 - Reconciliation #2772(timer) Kafka(default/my-cluster): Failed to reconcile
io.strimzi.operator.common.operator.resource.TimeoutException: Exceeded timeout of 300000ms while waiting for Ingress resource my-cluster-kafka-external-bootstrap in namespace event-center to be addressable`

and nginx ingress controller reported the following log:
`W1121 01:42:25.786854       6 controller.go:1112] Service "default/my-cluster-kafka-external-0" does not have any active Endpoint.`

it seems like operator is waitting for ingress controller set an address, but ingress controller is waiting for Strimzi start kafka. so Whether Strimzi needs a LoadBalancer when I configure external listener with ingress type? I just run my k8s on Bare-metal environment.

**Comment by scholzj** [comment, marked as answer]
My Kubernetes Nginx Ingress controller does not wait for the Kafka cluster to be running to confirm it accepted the Ingress resource and set its status. So if your does, it is probably something in your configuration.

**Reply by 1123183721** [reply]
Thanks for your replay. You are right, I must force nginx ingress controller publish status address with '--update-status' and '--publish-status-address'. Strimzi nerver start a kafka instance util ingress be addressable.
